### PR TITLE
Feature/image tests

### LINKED ISSUE #1833: Add tests to Cloudinary Image component
It is causing too much trouble not to have tests.

### PR DESCRIPTION
**Pull request by clari182**
- closes #1833

### CODE REVIEW

File: site/gatsby-site/cypress/e2e/integration/images/cloudinaryImages.cy.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                  }).then((resp) => {
+                    if (resp.status === 400 || resp.status === 404) {
+                      cy.wrap($el).should('have.class', 'hidden');
+                    }
+                  });
+                });
+            });
+          } else {
+            cy.log(`No images found on page, skipping test for path ${path}`);
+            it.skip(`Should display a placeholder image if image is failing on page ${path}`, () => {});
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/images/cloudinaryImages.cy.js:58]: I think this line is not necessary.

File: site/gatsby-site/cypress/e2e/integration/images/cloudinaryImages.cy.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          if (selectorExists) {
+            cy.get('[data-cy="cloudinary-image"]').each(($el) => {
+              cy.wrap($el)
+                .should('have.attr', 'src')
+                .then(($src) => {
+                  cy.request({
+                    url: $src.toString(),
+                    failOnStatusCode: false,
+                  }).then((resp) => {
+                    if (resp.status === 400 || resp.status === 404) {
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/images/cloudinaryImages.cy.js:50]: You should also make sure that the fallback is showing

File: site/gatsby-site/cypress/e2e/integration/images/cloudinaryImages.cy.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    '/entities/',
+    '/entities/facebook/',
+    '/blog/incident-report-2023-january/',
+    '/taxonomy/cset/',
+    '/summaries/wordcounts/',
+    '/about/', // doc template
+  ];
+
+  paths.forEach((path) => {
+    it(`Should not display broken images on page ${path}`, () => {
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/images/cloudinaryImages.cy.js:31]: Please move these tests to the `pages.cy.js`file to save some loading time.

**Review by cesarvarela — COMMENTED** [review]
I was thinking more about testing the component in different situations, like when the image loading fails to make sure it is rendering a fallback.

**Comment by clari182** [comment]
@cesarvarela ready for another review

File: site/gatsby-site/cypress/e2e/integration/pages.cy.js [hunk 4]
```diff
@@ -111,5 +111,44 @@ describe('Pages', () => {
         }
       });
     });
+
+    it(`Should not display broken images on page ${path}`, () => {
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/pages.cy.js:115]: Sorry, I wasn't very clear. Please move these tests to the `it` above so we run them all in one visit to reduce loading times (at least until we have more powerful boxes ).

(let's see how flaky this gets)

**Review by cesarvarela — CHANGES_REQUESTED** [review]

**Review by cesarvarela — APPROVED** [review]

**Pull request merged**